wagtail/api/v2/utils.py

```diff
@@ -53,3 +53,150 @@ def filter_page_type(queryset, page_models):
         qs |= queryset.type(model)
 
     return qs
+
+
+class FieldsParameterParseError(ValueError):
+    pass
+
+
+def parse_fields_parameter(fields_str):
+    """
+    Parses the ?fields= GET parameter. As this parameter is supposed to be used
+    by developers, the syntax is quite tight (eg, not allowing any whitespace).
+    Having a strict syntax allows us to extend the it at a later date with less
+    chance of breaking anyone's code.
+
+    This function takes a string and returns a list of tuples representing each
+    top-level field. Each tuple contains three items:
+     - The name of the field (string)
+     - Whether the field has been negated (boolean)
+     - A list of nested fields if there are any, None otherwise
+
+    Some examples of how this function works:
+
+    >>> parse_fields_parameter("foo")
+    [
+        ('foo', False, None),
+    ]
+
+    >>> parse_fields_parameter("foo,bar")
+    [
+        ('foo', False, None),
+        ('bar', False, None),
+    ]
+
+    >>> parse_fields_parameter("-foo")
+    [
+        ('foo', True, None),
+    ]
+
+    >>> parse_fields_parameter("foo(bar,baz)")
+    [
+        ('foo', False, [
+            ('bar', False, None),
+            ('baz', False, None),
+        ]),
+    ]
+
+    It raises a FieldsParameterParseError (subclass of ValueError) if it
+    encounters a syntax error
+    """
+
+    def get_position(current_str):
+        return len(fields_str) - len(current_str)
+
+    def parse_field_identifier(fields_str, is_first=False):
+        first_char = True
+        negated = False
+        ident = ""
+
+        while fields_str:
+            char = fields_str[0]
+
+            if char in ['(', ')', ',']:
+                if not ident:
+                    raise FieldsParameterParseError("unexpected char '%s' at position %d" % (char, get_position(fields_str)))
+
+                if ident == '*' and char == '(':
+                    # * cannot have nested fields
+                    raise FieldsParameterParseError("unexpected char '%s' at position %d" % (char, get_position(fields_str)))
+
```

Comment:
need to add a check in here to disallow negated fields with sub fields (eg ``-image(id)``)